backend/de.metas.handlingunits.base/src/main/java/de/metas/handlingunits/pporder/api/impl/AbstractPPOrderReceiptHUProducer.java

```diff
@@ -374,6 +383,21 @@ private void updateReceivedHUs(
 			{
 				huAttributes.setValue(AttributeConstants.ATTR_LotNumber, lotNumber);
 			}
+			else
+			{
+				final I_PP_Order_BOM ppOrderBom = ppOrderBOMDAO.getByOrderIdOrNull(ppOrderId);
```

Comment:
at least a part of this code chunk can go outside of for loop.
Maybe? you can consider having a method, which caches some of the results in class fields...